### DISCUSSION TITLE
Question: How to know whether a transaction succeeded or failed

### DISCUSSION THREAD

**Discussion by nir-peled** [prisma/prisma#23371] [Q&A] [answered]
### Question

I'm using Prisma with Accelerate in Next.js with TypeScript. 
One of my functions is making a transaction of [update, deleteMany, createMany] in sequential operations. 
I want the function to return whether the operations succeeded or not. 
How do I do that? I scoured the documentation but didn't found anything. Maybe I missed something obvious, I'll be glad if you could point me to it.

The relevant code is:
```ts
let result = await prisma.$transaction([
	prisma.race.update({ where: { id }, data }),
	prisma.raceContestant.deleteMany({ where: { id: { in: remove_contestants } } }),
	prisma.raceContestant.createMany({
		data: new_contestants.map((con) => ({
			race_id: id,
			...con,
		})),
	}),
]);
```
Thank you

### Information about Prisma Schema, Client Queries and Environment (optional)

```ts
// Add any relevant Prisma Client queries here
let result = await prisma.$transaction([
	prisma.race.update({ where: { id }, data }),
	prisma.raceContestant.deleteMany({ where: { id: { in: remove_contestants } } }),
	prisma.raceContestant.createMany({
		data: new_contestants.map((con) => ({
			race_id: id,
			...con,
		})),
	}),
]);
```

**Comment by ludralph** [comment, marked as answer]
Hi @nir-peled 👋 

Thank you for raising this question.

You are performing an `update`, `deleteMany`, and `createMany` operation within a transaction. The `result` variable will contain the results of these operations if they all succeed. If any operation fails, an error will be thrown and the transaction will be rolled back.
To determine if the operations succeeded, you can use a try-catch block. If an error is thrown, you can catch it and handle it accordingly. Here's an example:
```js
try {
  let result = await prisma.$transaction([
    prisma.race.update({ where: { id }, data }),
    prisma.raceContestant.deleteMany({ where: { id: { in: remove_contestants } } }),
    prisma.raceContestant.createMany({
      data: new_contestants.map((con) => ({
        race_id: id,
        ...con,
      })),
    }),
  ]);

  console.log('Transaction succeeded:', result);
} catch (error) {
  console.error('Transaction failed:', error);
}
```
In this example, if the transaction succeeds, a success message is logged to the console along with the results of the operations. If the transaction fails, an error message is logged to the console along with the error that was thrown.

The `result` variable in your code will be an array containing the results of each operation in the transaction. The order of the results in the array will match the order of the operations in the transaction.  The `result` array will contain the result of the `update` operation, the result of the `deleteMany` operation, and the result of the `createMany` operation, in that order. The `update` operation will return the updated record, the `deleteMany` operation will return an object with a `count` property, and the `createMany` operation will return an object with a `count` property.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by nir-peled** [reply]
Thank you! It might be a good idea to add this to the documentation

**Reply by ludralph** [reply]
Hi @nir-peled, we do have this documented in this section of the [documentation](https://www.prisma.io/docs/orm/prisma-client/queries/transactions#sequential-prisma-client-operations)
